spec/lib/bank_accounts/account_spec.rb

```diff
@@ -0,0 +1,60 @@
+require 'spec_helper'
+require 'bank_accounts'
+
+describe BankAccounts::Account do
+
+  context "When creating a bank account" do
+    it "responds to Account instantiation with 2 arguments" do
+      expect(BankAccounts::Account).to respond_to(:new).with(2).arguments
+    end
+
+    it "raises an error when an account is created with a negative balance" do
+      expect{ BankAccounts::Account.new(0, -1) }.to raise_error(ArgumentError, "Sorry, you can't start an account with a negative balance.")
+    end
+
+    it "rounds initial balance floats > longer than 2 decimal spots, down to 2" do
+      account = BankAccounts::Account.new(0, 10.2599)
+      expect(account.balance).to eq(10.25)
+    end
+  end
+
+  context "When withdrawing money from bank account" do
+    let(:account) { BankAccounts::Account.new(0, 200) }
+
+    it "returns an updated account balance when withdrawing money" do
+      expect(account.withdraw(10)).to eq(190)
+      expect(account.withdraw(40)).to eq(150)
+      expect(account.withdraw(36.2596)).to eq(113.74)
+    end
+
+    it "raises an error when the account is overdrawn" do
+      expect{ account.withdraw(500) }.to raise_error(ArgumentError, "INSUFFICIENT FUNDS\nYour current balance is $#{account.balance}.")
+    end
+  end
+
+  context "When depositing money into the bank account" do
+    let(:account) { BankAccounts::Account.new(0, 500) }
```

Comment:
Would it be possible to create an account object that could be placed at the `describe BankAccounts::Account` level instead of the context levels and cover all of the tests?